src/istio/control/http/attributes_builder.cc

```diff
@@ -128,12 +128,18 @@ void AttributesBuilder::ExtractAuthAttributes(CheckData *check_data) {
     builder.AddStringMap(utils::AttributeName::kRequestAuthClaims, payload);
   }
   std::string source_user;
-  if (check_data->GetSourceUser(&source_user)) {
+  if (check_data->GetPrincipal(true, &source_user)) {
     // TODO(diemtvu): remove kSourceUser once migration to source.principal is
     // over. https://github.com/istio/istio/issues/4689
     builder.AddString(utils::AttributeName::kSourceUser, source_user);
```

Comment:
while you are here, can you get rid of this?

Reply:
No. This is a terrible practice to make arbitrary changes in unrelated PRs.